### PR TITLE
Add SKIP

### LINKED ISSUE #25: Provide mechanism to skip a test
It would be good if Fiasco had a way that the test opted to skip itself. Example:

```
(fiasco:deftest test-socket ()
  (when os-is-windows
    (fiasco:skip-test))
  ...)
```

Prefer not to use `#-windows` as I think that's bad practice.

### PR DESCRIPTION
**Pull request by notmgsk**
First pass at implementing method for skipping tests.

Closes #25

### CODE REVIEW

File: src/suite.lisp [hunk 1]
```diff
@@ -213,15 +213,20 @@ See RUN-TESTS for the meaning of the remaining keyword arguments."
...
                            (run-test-body-in-handlers test function)))
-           (failures (failures-of *context*)))
+           (failures (failures-of *context*))
+           (skipped (skipped *context*)))
       (unless (suite-p)
         (format *pretty-log-stream* "~v@{~C~:*~}"
                 (max 1 (- *test-progress-print-right-margin*
                           (output-column *pretty-log-stream*)
-                          (length "[XXXX]")))
+                          (length "[SKIPPED]")))
```
Comment (joaotavora) [inline src/suite.lisp]: Maybe "SKIP" would be enough here. It's not very grammatically correct, but then neither is "FAIL" :-)

**Comment by joaotavora** [comment]
@notmgsk can you summarize what is already done and missing in this PR? As I see it, the only thing that's missing is a bit of documentation in the README, perhaps just an example will do, as the UI is easy enough to figure out.

**Comment by notmgsk** [comment]
@joaotavora I applied your changes, and added a brief part to the README. Let me know if you think there's anything else to consider.

**Comment by joaotavora** [comment]
So I squashed and merged your PR, noticed a couple of problems when running via `run-package-tests` fixed them, and... broke it again!

...because I though it wouldn't be fair to name this "fiasco" without living up to the name daily...

**Comment by notmgsk** [comment]
> So I squashed and merged your PR, noticed a couple of problems when running via `run-package-tests` fixed them, and... broke it again!
> 
> ...because I though it wouldn't be fair to name this "fiasco" without living up to the name daily...

lol! Good catch :)
